.github/workflows/CI-system-test.yml

```diff
@@ -0,0 +1,65 @@
+name: System Tests and Publish Docker Image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - feat/web3-943-post-merge-pipeline
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+-[0-9]+.[0-9]+.[0-9]+*"
+
+permissions:
+  contents: read
+
+jobs:
+  execute-system-tests:
+    name: Execute System Tests
+    uses: HorizenLabs/zkVerify-qa/.github/workflows/CI-e2e-tests.yml@feat/always-build
+    with:
+      zkverify_version: ${{ startsWith(github.ref, 'refs/tags/') && github.ref_name || github.ref_name }}
+    secrets:
+      DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+      DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+      QA_SLACK_WEBHOOK_URL: ${{ secrets.QA_SLACK_WEBHOOK_URL }}
+      GH_TOKEN: ${{ secrets.GH_TOKEN }}
+
+  build-and-publish:
+    name: Build and Publish Docker Image
+    runs-on: ubuntu-latest
+    needs: execute-system-tests # Wait for e2e tests to finish
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0 # Necessary for git operations in setup_env.sh
+
+      - name: Get Short Commit Hash
+        id: get_short_commit_hash
+        run: |
+          echo "short_commit_hash=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Set up Environment Variables and Build
+        env:
+          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
+          DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+          PRIVATE_DOCKER_REPO: horizenlabs/zkverify-dev
+          GITHUB_REF_NAME: ${{ github.ref_name }}
+          COMMIT_HASH: ${{ steps.get_short_commit_hash.outputs.short_commit_hash }}
+
+        run: |
+          # Source the setup script to set up environment variables
+          source "${GITHUB_WORKSPACE}/ci/setup_env.sh"
+
+          # Overwrite environment variables if building from main branch
+          if [[ "${{ startsWith(github.ref, 'refs/heads/feat/web3-943-post-merge-pipeline') }}" == 'true' ]]; then
+            export IS_A_RELEASE="false"
+            export DEV_RELEASE="true"
+            export PROD_RELEASE="false"
+            export TEST_RELEASE="false"
+          fi
```

Comment:
Maybe we should consider this a "test release"

- Prod release is from a production release formatted tag (X.Y.Z), published to production docker repo
- Dev release is from a dev release formatted tag (X.Y.Z-rc), published to production docker repo
- Test release is anything else, published to dev docker repo

@ysibirski Thoughts on this?

Reply:
Yeah, I think we can def consider it since TEST_RELEASE is currently 

```
^[0-9]+\.[0-9]+\.[0-9]+\-[0-9]+\.[0-9]+\.[0-9]+-[a-zA-Z0-9]+$
```
For example: `1.1.1-1.2.3-yuri`